robotpy_build/pkgcfg_provider.py

```diff
@@ -117,13 +117,35 @@ class PkgCfgProvider:
 
     def __init__(self):
         self.pkgs = {}
-        for entry_point in iter_entry_points(group="robotpybuild", name=None):
-            try:
-                pkg = PkgCfg(entry_point)
-            except Exception as e:
-                warnings.warn(f"Error loading entry point {entry_point.name}: {e}")
-            else:
-                self.add_pkg(pkg)
+
+    def detect_pkgs(self):
+
+        entry_points = list(iter_entry_points(group="robotpybuild", name=None))
```

Comment:
Why is this list conversion necessary?

Reply:
It's there because `entry_points` is iterated over multiple times in `ingest()`.
My logic there was (for example): 
rev explicitly depends on wpilib but doesn't mention ntcore. wpilib explicitly depends on ntcore. On the first pass (of ingest), wpilib would be loaded but ntcore would not, since it is not listed as a dependency. On the second pass, ntcore could be loaded as it's now in the list of dependencies (via wpilib).
I don't know if this logic is necessary though.